### PR TITLE
SAK-39937 Swapped event assignment id and submission id for full refe…

### PR DESCRIPTION
**Pull request by adrianfish**
…rence

### CODE REVIEW

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -5200,12 +5200,14 @@ public void doView_submission(RunData data) {
             if (submission != null) {
                 // submission read event
             	LRS_Statement statement = getStatementForViewSubmittedAssignment(submission.getId(), a.getTitle());
-                Event event = eventTrackingService.newEvent(AssignmentConstants.EVENT_ACCESS_ASSIGNMENT_SUBMISSION, submission.getId(), null, false, NotificationService.NOTI_OPTIONAL, statement);
+                String ref = AssignmentReferenceReckoner.newAssignmentReferenceReckoner(null, submission, null, null, null, null, null).getReference();
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: String ref = AssignmentReferenceReckoner.reckoner().submission(submission).reckon().getReference();

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 2]
```diff
@@ -5200,12 +5200,14 @@ public void doView_submission(RunData data) {
...
+                Event event = eventTrackingService.newEvent(AssignmentConstants.EVENT_ACCESS_ASSIGNMENT_SUBMISSION, ref, null, false, NotificationService.NOTI_OPTIONAL, statement);
                 eventTrackingService.post(event);
             } else {
                 // otherwise, the student just read assignment description and prepare for submission
             	LRS_Statement statement = getStatementForViewAssignment(a.getId(), a.getTitle());
-                Event event = eventTrackingService.newEvent(AssignmentConstants.EVENT_ACCESS_ASSIGNMENT, a.getId(), null, false, NotificationService.NOTI_OPTIONAL, statement);
+                String ref = AssignmentReferenceReckoner.newAssignmentReferenceReckoner(a, null, null, null, null, null, null).getReference();
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: String ref = AssignmentReferenceReckoner.reckoner().assignment(a).reckon().getReference();

Comment (adrianfish) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Cool, ta. I'll sort it now.

**Comment by adrianfish** [comment]
@ern Pushed

**Pull request merged**